language/move-prover/boogie-backend/src/bytecode_translator.rs

```diff
@@ -181,7 +181,13 @@ impl<'env> BoogieTranslator<'env> {
         for idx in &mono_info.type_params {
             let param_type = boogie_type_param(env, *idx);
             let suffix = boogie_type_suffix(env, &Type::TypeParameter(*idx));
-            emitln!(writer, "type {};", param_type);
+            emitln!(writer, "type {{:datatype}} {};", param_type);
+            emitln!(
+                writer,
+                "function {{:constructor}} {}($id: $2_object_UID): {};",
```

Comment:
This will cause a compilation error when there is no mention of `sui::object::UID` in the code.

A quick (and hacky) solution is to test whether `mono_info.structs` contains `object::UID` and if so, emit the two lines, otherwise, leave things unchanged.